### PR TITLE
DDF-3468 Fix for notifications

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Fixes error displaying UI notifications.

#### Who is reviewing it? 
@adimka 
@mackncheesiest 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams/ui

#### Choose 2 committers to review/merge the PR. 
@andrewkfiedler
@millerw8

#### How should this be tested? (List steps with links to updated documentation)
Create a query and turn on notifications (bell icon by query), and select the interval.
Inject new data that matches the query.
Wait up to interval.
Notice that the notification bell in the upper right indicates new data found.

#### Any background context you want to provide?
typeof check is necessary as sometimes the response is just "successful" and other times it's a JSON object.

#### What are the relevant tickets?
[DDF-3468](https://codice.atlassian.net/browse/DDF-3468)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
For the general case, merging is gated on the following:

* Review by at least two contributors
* Review by at least two committers
* Successful CI
* Successful static analysis (SonarQube/Coverity/etc.)
* Heroing of change (details vary by ticket)

Certain exceptions may occur. For example, if there is a problem with the CI environment then a manually performed full build and static analysis check may be substituted _at the discretion of the repository maintainers_. Additionally, repository metadata such as this `PULL_REQUEST_TEMPLATE`, the repository `README`, and other non-functional items may be merged without requiring CI and static analysis.

Finally, an abbreviated review process may be in place for forward-port tickets immediately following branching and prior to branch divergence. In that case, approvals, CI, static analysis, and heroing that occur against the PR against the _release branch_ can be considered sufficient for reviewing the PR against the _master branch_.

Once the branches diverge, the process will return to normal.

Insofar as divergence is not a function of time but of change to the parent branch, the repository maintainers will determine when abbreviated review ends.

When abbreviated review is in effect, a note to that effect will be at the top of this template. Following is an example note:
```
##### ABBREVIATED REVIEW BETWEEN 2.12.X AND MASTER IS IN EFFECT
____

```

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by clockard** [comment]
should this be ported to the 2.11.x release branch?

**Comment by troymohl** [comment]
@clockard Once this is approved I will port to 2.11.x

File: catalog/ui/catalog-ui-search/src/main/webapp/js/QueryPolling.js [hunk 1]
```diff
@@ -132,8 +132,8 @@ define([
 
     checkForFailures = function(responses, originalQuery, timeRange) {
         _.forEach(responses, function(response) {
-            if (response[1] === "error" || !response[0].status.successful) {
-                var srcId = JSON.parse(response[0].options.data).src;
+            if (response === "error" ||  (typeof response.status !== 'undefined' && !response.status.successful)) {
```
Comment (andrewkfiedler) [inline catalog/ui/catalog-ui-search/src/main/webapp/js/QueryPolling.js]: At the very least this should be:

```
if (Array.isArray(response)) {
     if (response[1] === "error" || (response[0].status !== undefined && !response[0].status.successful)) {
          const srcId = JSON.parse(response[0].options.data).src;
          addFailure(srcId, originalQuery, timeRange);
     }
} else if (response.constructor === Object) {
     if (response.status !== undefined && !response.status.successful) {
          const srcId = JSON.parse(response.responseJSON.options.data).src;
          addFailure(srcId, originalQuery, timeRange);
     }
} else if (response === 'error') {
     // if this happens we need to find a way to retry without knowing src id
}
```

**Comment by troymohl** [comment]
@andrewkfiedler I have updated the pull request to match the snippet you provided.

To reproduce this you need to select `Disable Cache` in the UI configuration.

**Comment by andrewkfiedler** [comment]
I was able to track down what I think is the source of the issue.  I put up a fix here: https://github.com/codice/ddf/pull/2716

Could you double check that it solves your issue?

**Pull request closed without merging**